### DISCUSSION TITLE
3D mouse always activates Real-Time (BUG)

### DISCUSSION THREAD

**Discussion by Nekbeth** [microdee/OpenUnrealSpaceMouse#36] [Q&A]
Hi,

Is there a way to disable the Real-Time checkmark when using this plug-in?  

Also the Menu and Fit buttons change the speed of the camera.

Thanks

**Comment by microdee** [comment]
for some reason I commented out the code which would turn realtime back off if it was off originally. I fixed it and it only turns on realtime while moving with the spacemouse, then turns it back off when finished moving. Of course it only does it when realtime was originally turned off by the user. If realtime would be kept off while moving, you wouldn't see the movement and it would be confusing, and I would receive bug reports saying "camera doesn't move when realtime turned off".

For Menu and Fit: This plugin doesn't know about the Legend of the buttons on the device (yet). Only the button position is known in the HID report, of which Menu and Fit are the first two (on SpaceMouse Pro and I presume on Enterprise at least). The reason 0 and 1 is the default buttons for the speed increase/decrease is that Space Navigator users can still use that feature out of the box. **BUT** you can configure this in `Editor Preferences -> Plugins -> SpaceMouse -> Button Mapping`. If the checkboxes are ticked, it's in learning mode and you can just press the button on the device to assign it to a function or a simulated shortcut.

**Reply by Nekbeth** [reply]
"If realtime would be kept off while moving, you wouldn't see the movement and it would be confusing, and I would receive bug reports saying "camera doesn't move when realtime turned off".

REPLY 
- I understand. Currently, if I set RealTime off in the Viewport and then touch my Spacemouse to navigate, it immediately turns it ON and stays ON.  There is no really a way to turn it OFF.  This makes this plugin incredibly performance expensive for the CPU/GPU to render (even turns on my fans).  A lot of us never use RealTime because of that,  we only activate it when your into Game mode or want to see some particle effects, other than that, RealTime should always be off. 

Also, I've tested previous plugins (not supported anymore) for UE4  and successfully moving the camera without messing with RealTime.  Here is the link, unfortunately it won't support current UE4 versions. 

https://github.com/ParcelRot/SpaceNav3D

Will this bug be fixed in future versions? 

Thanks

**Reply by microdee** [reply]
I made a hotfix release, I'm on a bigger refactoring again that can take a while. Please try https://github.com/microdee/UE4-SpaceMouse/releases/tag/1.1.4 Marketplace release is coming couple days later.

**Reply by Nekbeth** [reply]
Cool, I'll try it out. Thanks

**Reply by aknittel1** [reply]
Your releases for 5.0 have regressed your hotfix. Now what @Nekbeth says is back: "if I set RealTime off in the Viewport and then touch my Spacemouse to navigate, it immediately turns it ON and stays ON. There is no really a way to turn it OFF. This makes this plugin incredibly performance expensive for the CPU/GPU to render (even turns on my fans). A lot of us never use RealTime because of that, we only activate it when your into Game mode or want to see some particle effects, other than that, RealTime should always be off."
